### PR TITLE
Consume "supervisor" field

### PR DESCRIPTION
**Pull request by faf0-addepar**
Fixes https://github.com/JupiterOne/graph-bamboohr/issues/11

### CODE REVIEW

**Comment by faf0-addepar** [comment]
@austinkelleher @ndowmon

File: src/steps/__snapshots__/index.test.ts.snap [hunk 1]
```diff
@@ -50,6 +50,7 @@ Object {
               "preferredName": null,
               "workEmail": "dwight@efficientoffice.com",
               "workPhone": "604-555-3134",
+              "supervisor": null,
```
Comment (ndowmon) [inline src/steps/__snapshots__/index.test.ts.snap:53]: @faf0-addepar according to the [test failure](https://github.com/JupiterOne/graph-bamboohr/runs/1477069727?check_suite_focus=true) this field ought to be `undefined` and not `null`... when I run `yarn test -u` it updates these fields to `undefined` for me... not sure why your setup is setting this value to `null` 🤷

Comment (ndowmon) [reply src/steps/__snapshots__/index.test.ts.snap:53]: It also looks like in the snapshot, these fields are added to the `rawData.employeeDetails` property, but not as first-class properties to the entity. When I run `yarn test -u`, the `supervisor` property is added to the entity.

Not sure why it woudn't be working for you. Maybe a local Jest config? If you can't get it to work, I can send a snapshot that you can commit in your branch.

Comment (faf0-addepar) [reply src/steps/__snapshots__/index.test.ts.snap]: Thanks @ndowmon , I manually added the snapshot data at first. I just reverted the changes and committed those resulting from `yarn test -u`. Now build is green :-)

**Review by ndowmon — APPROVED** [review]
Looks great! Thanks for adding the `supervisor` property, @faf0-addepar.

Based on this change, I wonder if it would be worth adding a relationship between employees and supervisors in the future. I'd be happy to collaborate again if it would help your J1 use cases!

**Pull request merged**